BUG: [unit test] pandas/io/clipboard/__init__.py:300: PyperclipException

### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.


### Reproducible Example

```python
# pytest pandas/tests/io/
============================================================================================ short test summary info =============================================================================================
FAILED pandas/tests/io/test_clipboard.py::TestClipboard::test_raw_roundtrip[\U0001f44d...] - pandas.errors.PyperclipException: 
FAILED pandas/tests/io/test_clipboard.py::TestClipboard::test_raw_roundtrip[\u03a9\u0153\u2211\xb4...] - pandas.errors.PyperclipException: 
FAILED pandas/tests/io/test_clipboard.py::TestClipboard::test_raw_roundtrip[abcd...] - pandas.errors.PyperclipException: 
========================================================================================= 3 failed, 232 passed in 1.55s ==========================================================================================
```


### Issue Description

The unit-test  https://github.com/pandas-dev/pandas/blob/047c11d7801ffd0cf679d606293af510e34d7b92/pandas/tests/io/test_clipboard.py#L398
does not pass.

I am using VSCode with the .devcontainer file (thus in a container), with a linux distribution (pop-os). 
Current commit I am on is 047c11d7801ffd0cf679d606293af510e34d7b92

I tried to do `apt-get install xclip` but it did not solve the issue. 

### Expected Behavior

The test should pass.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 047c11d7801ffd0cf679d606293af510e34d7b92
python           : 3.8.13.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.8.0-7630-generic
Version          : #32~1609193707~20.10~781bb80-Ubuntu SMP Tue Jan 5 21:29:56 UTC 2
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : C.UTF-8
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.6.0.dev0+89.g047c11d780.dirty
numpy            : 1.22.4
pytz             : 2022.2.1
dateutil         : 2.8.2
setuptools       : 65.3.0
pip              : 22.2.2
Cython           : 0.29.32
pytest           : 7.1.3
hypothesis       : 6.54.5
sphinx           : 4.5.0
blosc            : None
feather          : None
xlsxwriter       : 3.0.3
lxml.etree       : 4.9.1
html5lib         : 1.1
pymysql          : 1.0.2
psycopg2         : 2.9.3
jinja2           : 3.0.3
IPython          : 8.5.0
pandas_datareader: 0.10.0
bs4              : 4.11.1
bottleneck       : 1.3.5
brotli           : 
fastparquet      : 0.8.3
fsspec           : 2021.11.0
gcsfs            : 2021.11.0
matplotlib       : 3.5.3
numba            : 0.55.2
numexpr          : 2.8.3
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : 0.17.8
pyarrow          : 9.0.0
pyreadstat       : 1.1.9
pyxlsb           : 1.0.9
s3fs             : 2021.11.0
scipy            : 1.9.1
snappy           : 
sqlalchemy       : 1.4.41
tables           : 3.7.0
tabulate         : 0.8.10
xarray           : 2022.6.0
xlrd             : 2.0.1
xlwt             : 1.3.0
zstandard        : 0.18.0
tzdata           : None

</details>

@mroeschke it is a `PyperclipException`. Strangely enough, I can make the test pass in another shell on the same docker machine. What am I missing?

### Stacktrace
```python
self = <pandas.tests.io.test_clipboard.TestClipboard object at 0x7f479a61ca00>, data = '👍...'

    @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
    @pytest.mark.xfail(
        reason="Flaky test in multi-process CI environment: GH 44584",
        raises=AssertionError,
        strict=False,
    )
    def test_raw_roundtrip(self, data):
        # PR #25040 wide unicode wasn't copied correctly on PY3 on windows
>       clipboard_set(data)

pandas/tests/io/test_clipboard.py:400: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/io/clipboard/__init__.py:640: in lazy_load_stub_copy
    return copy(text)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pandas.io.clipboard.init_no_clipboard.<locals>.ClipboardUnavailable object at 0x7f479a3c3460>, args = ('👍...',), kwargs = {}

    def __call__(self, *args, **kwargs):
>       raise PyperclipException(EXCEPT_MSG)
E       pandas.errors.PyperclipException: 
E           Pyperclip could not find a copy/paste mechanism for your system.
E           For more information, please visit
E           https://pyperclip.readthedocs.io/en/latest/#not-implemented-error

pandas/io/clipboard/__init__.py:300: PyperclipException
```


# Difference in execution: 
Strangely enough, I can make that test past of fail based on where I execute the notebook... 
I would be quite please to understand what makes the difference between the first screenshot and the second one...
The only one I found is that one the left pannel I have a box for the shell where the test fail, and a console logo for the test that pass. 
![image](https://user-images.githubusercontent.com/6614121/189647351-e3559ad4-dbd7-469c-8d46-8af2480d4329.png)



![image](https://user-images.githubusercontent.com/6614121/189643955-d153b1e8-d050-4fdc-95bc-216cc97a9cc2.png)
![image](https://user-images.githubusercontent.com/6614121/189647725-3d4f9c58-2160-46bc-b42b-f1a108632ab9.png)

--- a/pandas/tests/io/test_clipboard.py
+++ b/pandas/tests/io/test_clipboard.py
@@ -391,6 +391,7 @@ class TestClipboard:
         self.check_round_trip_frame(df, encoding=enc)
 
     @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
+    @pytest.mark.single_cpu
     @pytest.mark.xfail(
         os.environ.get("DISPLAY") is None,
         reason="Cannot be runed if a headless system is not put in place with Xvfb",

--- a/pandas/tests/io/test_clipboard.py
+++ b/pandas/tests/io/test_clipboard.py
@@ -390,8 +390,8 @@ class TestClipboard:
     def test_round_trip_valid_encodings(self, enc, df):
         self.check_round_trip_frame(df, encoding=enc)
 
-    @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
     @pytest.mark.single_cpu
+    @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
     @pytest.mark.xfail(
         os.environ.get("DISPLAY") is None,
         reason="Cannot be runed if a headless system is not put in place with Xvfb",

--- a/pandas/tests/io/test_clipboard.py
+++ b/pandas/tests/io/test_clipboard.py
@@ -1,5 +1,5 @@
-from textwrap import dedent
 import os
+from textwrap import dedent
 
 import numpy as np
 import pytest
@@ -392,8 +392,8 @@ class TestClipboard:
 
     @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
     @pytest.mark.xfail(
-        os.environ.get('DISPLAY') is None, 
-        reason="Cannot be runned if a headless system is not put in place with ",
+        os.environ.get("DISPLAY") is None,
+        reason="Cannot be runed if a headless system is not put in place with Xvfb",
         strict=True,
     )
     def test_raw_roundtrip(self, data):

--- a/pandas/tests/io/test_clipboard.py
+++ b/pandas/tests/io/test_clipboard.py
@@ -1,4 +1,5 @@
 from textwrap import dedent
+import os
 
 import numpy as np
 import pytest
@@ -391,9 +392,9 @@ class TestClipboard:
 
     @pytest.mark.parametrize("data", ["\U0001f44d...", "Ωœ∑´...", "abcd..."])
     @pytest.mark.xfail(
-        reason="Flaky test in multi-process CI environment: GH 44584",
-        raises=AssertionError,
-        strict=False,
+        os.environ.get('DISPLAY') is None, 
+        reason="Cannot be runned if a headless system is not put in place with ",
+        strict=True,
     )
     def test_raw_roundtrip(self, data):
         # PR #25040 wide unicode wasn't copied correctly on PY3 on windows

